src/figuresClasses.ts

```
@@ -1,11 +1,95 @@
-export interface Figure {}
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
+type ColorType = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: ShapeType = 'triangle';
+
+  color: ColorType;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: ColorType, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be positive numbers.');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('These sides cannot form a triangle.');
```

This violates checklist item: 'Error messages thrown in constructors must clearly describe the specific validation failure (e.g., "Width must be greater than 0", "Sides 1, 2 and 3 can't form a triangle")'.

The error message 'These sides cannot form a triangle.' is not specific about which sides or values are invalid. You should throw an error like 'Triangle sides {a}, {b}, {c} do not satisfy the triangle inequality'.